homeassistant/components/weatherflow/sensor.py

```diff
@@ -348,6 +369,22 @@ def __init__(
                 description.imperial_suggested_unit
             )
 
+    @property
+    def icon(self) -> str | None:
+        """Get icon."""
+
+        value = (
+            int(self.native_value)
+            if self.native_value is not None
+            and self.native_value != "none"
+            and isinstance(self.native_value, (int, float, str))
+            else None
+        )
+
+        if value and self.entity_description.icon_fn is not None:
```

Comment:
We should do a None check on icon_fn first and immediately return to avoid unnecessary steps.

Reply:
Fair enough... :)